warehouse/legacy/api/pypi.py

```diff
@@ -786,11 +807,12 @@ def file_upload(request):
         if filename.endswith(".whl"):
             wheel_info = _wheel_file_re.match(filename)
             plats = wheel_info.group("plat").split(".")
-            if set(plats) - ALLOWED_PLATFORMS:
-                raise _exc_with_message(
-                    HTTPBadRequest,
-                    "Binary wheel for an unsupported platform.",
-                )
+            for plat in plats:
+                if not _valid_platform_tag(plat):
+                    raise _exc_with_message(
+                        HTTPBadRequest,
+                        "Binary wheel for an unsupported platform.",
```

Comment:
I think it would be helpful for the users to report the offending paltform tag(s) in the exception message.